frontend/src/metabase/components/UserAvatar/UserAvatar.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+
+import MetabaseUtils from "metabase/lib/utils";
+import { Avatar, AvatarProps } from "./UserAvatar.styled";
+
+interface UserAvatarProps extends AvatarProps {
+  user: User;
+}
+
+interface GroupProps {
+  user: Group;
+}
+
+interface User {
+  first_name: string | null;
+  last_name: string | null;
+  common_name: string;
+  email?: string;
+}
+
+interface Group {
+  first_name: string;
+}
+
+export default function UserAvatar({
+  user,
+  ...props
+}: UserAvatarProps | GroupProps) {
+  return <Avatar {...props}>{userInitials(user) || "?"}</Avatar>;
+}
+
+function initial(name?: string | null) {
+  return typeof name === "string" ? name.charAt(0).toUpperCase() : "";
+}
+
+function userInitials(user: User | Group) {
+  if (user) {
+    return nameInitials(user) || emailInitials(user as User);
+  }
+
+  return null;
+}
+
+function nameInitials(user: User | Group) {
+  if ("common_name" in user) {
+    return initial(user.first_name) + initial(user.last_name);
+  }
+
+  // render group
+  return initial(user.first_name);
+}
+
+function emailInitials(user: User) {
+  if (user.email && MetabaseUtils.isEmail(user.email)) {
```

Comment:
Need a guard check for `user.email` otherwise, line 56 will complain that `user.email` could be undefined. This is because on segment and metric revisions, there would be no `email` returned.

Reply:
Would it be enough to check for `user.email` only? I believe it should be either email or undefined

Reply:
Thank you! Fixed.